### PR TITLE
Sidebar settings

### PR DESCRIPTION
**Pull request by Nomehh**
I noticed that there was no functionality to open or close the explorer by default.

To address this, I implemented a feature that allows users to choose whether the explorer is extended or folded by default, through the settings. 
Please let me know if it suits to you.
(By default, it's still opened) 

image

### CODE REVIEW

File: projects/nge-ide/src/sidebar/sidebar.component.ts [hunk 1]
```diff
@@ -76,10 +77,12 @@ export class SidebarComponent implements OnInit, OnDestroy {
     private readonly ideService: IdeService,
     private readonly storageService: StorageService,
     private readonly changeDetectionRef: ChangeDetectorRef,
-    private readonly viewContainerService: ViewContainerService
+    private readonly viewContainerService: ViewContainerService,
+    private readonly settingsService: SettingsService
   ) {}
 
   ngOnInit(): void {
+    this.settingsService.activate();
```
Comment (mciissee) [inline projects/nge-ide/src/sidebar/sidebar.component.ts:85]: Contributions should not be activated manually, this is handled by the ide service upon startup : 

https://github.com/cisstech/nge-ide/blob/main/projects/nge-ide/core/src/ide.service.ts#L82

File: projects/nge-ide/src/sidebar/sidebar.component.ts [hunk 2]
```diff
@@ -110,6 +113,7 @@ export class SidebarComponent implements OnInit, OnDestroy {
   ngOnDestroy(): void {
     this.saveState();
     this.subscriptions.forEach((s) => s.unsubscribe());
+    this.settingsService.deactivate();
```
Comment (mciissee) [inline projects/nge-ide/src/sidebar/sidebar.component.ts:116]: Same remark here

**Review by mciissee — APPROVED** [review]
@Nomehh Thanks for the PR, it's a nice feature, I will merge it after the resolving of the 2 comments above 👌🏽

File: projects/nge-ide/src/sidebar/sidebar.component.ts (hunk 1 above)
Comment (Nomehh) [reply projects/nge-ide/src/sidebar/sidebar.component.ts]: Thanks for the review! I'm fixing it right now.

**Comment by Nomehh** [comment]
Contributions are no longer manually activated; instead, I restore the state of the sidebar after the contributions have been added.

**Comment by Nomehh** [comment]
Last commit is to prevent this case 

Capture d’écran 2024-06-14 à 17 31 13

**Comment by mciissee** [comment]
Thank for the fixes @Nomehh. A new version will be available on npm in few minutes

**Comment by mciissee** [comment]
https://www.npmjs.com/package/@cisstech/nge-ide/v/17.7.3

**Pull request merged**